openforcefield/topology/molecule.py

```diff
@@ -35,16 +35,15 @@
 #=============================================================================================
 
 import numpy as np
-from copy import deepcopy
-
-from distutils.spawn import find_executable
+from collections import OrderedDict
 
 from simtk import unit
 from simtk.openmm.app import element
 
 import openforcefield
-from openforcefield.utils.toolkits import OPENEYE_INSTALLED, RDKIT_INSTALLED, TOOLKIT_PRECEDENCE, SUPPORTED_FILE_FORMATS
-from openforcefield.utils.toolkits import requires_rdkit, requires_openeye
+from openforcefield.utils.toolkits import OPENEYE_AVAILABLE, RDKIT_AVAILABLE, GLOBAL_TOOLKIT_REGISTRY
```

Comment:
Hopefully, we can go through the toolkit wrappers for all our cheminformatics needs in `molecule.py`, so we don't actually need to know which toolkit is servicing our needs (and don't need this here).

Reply:
I agree with this. The one tricky part is [1] that our Molecule constructor wants to type-check for OEMol and RDMol-type inputs _here in the Molecule class_. My thought is that we could have the toolkits each provide a "from_object" function that will do the type checking for an input and return False it if it's not compatible, or the results of from_rdkit or from_openeye (or whatever other toolkits we add later) if it is compatible. Then the Molecule constructor could iterate through our toolkit_registry and to run from_object using each one, until we either get a Molecule out or they've all returned False.

[1] https://github.com/openforcefield/openforcefield/pull/119/files/11b71c413a5f1487e364085fdaa28f6675d3fd6f#diff-dcc45fa83688f86668e384f78c5ab22aR1223

Reply:
This sounds like a good solution!